skorch/tests/test_utils.py

```diff
@@ -417,3 +417,72 @@ def type_truth_table():
         type_truth_table())
     def test_data_types(self, is_skorch_dataset, input_data, expected):
         assert is_skorch_dataset(input_data) == expected
+
+
+class TestFilterRequiresGrad():
+
+    @pytest.fixture
+    def filter_requires_grad(self):
+        from skorch.utils import filter_requires_grad
+        return filter_requires_grad
+
+    def test_all_parameters_requires_graident(
+            self, filter_requires_grad):
+        pgroups = [{
+            'params': [torch.zeros(1, requires_grad=True),
+                       torch.zeros(1, requires_grad=True)],
+            'lr': 0.1
+        }, {
+            'params': [torch.zeros(1, requires_grad=True)]
+        }]
+
+        filter_pgroups = list(filter_requires_grad(pgroups))
+        assert len(filter_pgroups) == 2
+        assert len(list(filter_pgroups[0]['params'])) == 2
+        assert len(list((filter_pgroups[1]['params']))) == 1
+
+        assert filter_pgroups[0]['lr'] == 0.1
+
+    def test_some_params_requires_graident(
+            self, filter_requires_grad):
+        pgroups = [{
+            'params': [
+                torch.zeros(1, requires_grad=True),
+                torch.zeros(1, requires_grad=True)
```

Comment:
I think we should test wether the filter function does not simply drop whole groups where only one element is filtered. We could do this by setting requires_grad to False here as well.

Reply:
I agree. PR was updated with this test.

Reply:
Where is that tested?

Reply:
Ah I misunderstood the comment, I updated this `test_some_params_requires_gradient` to reflect this.